### DISCUSSION TITLE
error: redefinition /previous definition of 'class HTTPClient'

### DISCUSSION THREAD

**Discussion by frogg1ven** [jgromes/RadioLib#301] [General]
Hi,
I'm using your library in my project along with ESP8266HTTPClient library, and there is colision between them, that I can't fix.
I have this errors:
`...\RadioLib-master\src/protocols/HTTP/HTTP.h:18:7: error: redefinition of 'class HTTPClient'
 class HTTPClient {
       ^
...\ESP8266HTTPClient\src/ESP8266HTTPClient.h:163:7: error: previous definition of 'class HTTPClient'
 class HTTPClient
       ^
`
I have read, that libraries need something like this:
`#ifndef libName_h
#define libName_h
class HTTPClient{
...
}
#endif
`
and I tried to fix this, but it doesn't seem to work, and both libraries has similar things already defined.
Could you help me with this problem?

**Comment by jgromes** [comment]
What you're referring to is called a macro guard, and RadioLib has those:

https://github.com/jgromes/RadioLib/blob/9ac6503bffa6ca578f4113c0554537a9a5dc0db6/src/protocols/HTTP/HTTP.h#L1-L2

However, the actual issue is a clash of the same name for different classes - both RadioLib and ESP8266HTTPClient have class named `HTTPClient`.

Fortunately, RadioLib has quite a clever way to exclude drivers for different modules. I excluded HTTP support from RadioLib on ESP8266 platform in the latest commit (since HTTP support in RadioLib is quite basic and likely offers no advantages over ESP8266HTTPClient), so it works now.

**Reply by frogg1ven** [reply]
thanks for help. it works now!

**Reply by HamzaHajeir** [reply]
Thanks @jgromes, I just had this error, I'm wondering when this would be released (as library version)? 

Thanks

**Reply by jgromes** [reply]
I guess there's no reason to not release, it's been a while and it looks like there's no major changes coming in the near future.
